dataset/src/main/scala/frameless/TypedColumn.scala

```diff
@@ -15,21 +15,28 @@ sealed class TypedColumn[T, U](
 )(
   implicit
   val uencoder: TypedEncoder[U]
-) extends UntypedExpression[T] {
+) extends UntypedExpression[T] { self =>
 
   def this(column: Column)(implicit uencoder: TypedEncoder[U]) {
     this(FramelessInternals.expr(column))
   }
 
   def untyped: Column = new Column(expr)
 
-  def ===(other: U): TypedColumn[T, Boolean] = {
-    new TypedColumn[T, Boolean](untyped === other)
-  }
+  def ===(other: U): TypedColumn[T, Boolean] = new TypedColumn[T, Boolean](self.untyped === other)
```

Comment:
Could you also bring in [Spark's Scaladoc](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/Column.scala) for these operations here?